postgres/lib/testcontainers/postgres.rb

```diff
@@ -0,0 +1,122 @@
+require_relative "postgres/version"
+require "testcontainers"
+require "uri"
+
+module Testcontainers
+  # PostgresContainer class is used to manage containers that runs a PostgresQL database
+  #
+  # @attr_reader [String] port used by the container
+  # @attr_reader [String] username used by the container
+  # @attr_reader [String] password used by the container
+  # @attr_reader [String] database used by the container
+  class PostgresContainer < ::Testcontainers::DockerContainer
+    # Default port used by the container
+    POSTGRES_DEFAULT_PORT = 5432
+
+    # Default image used by the container
+    POSTGRES_DEFAULT_IMAGE = "postgres:latest"
+
+    POSTGRES_DEFAULT_USERNAME = "test"
+    POSTGRES_DEFAULT_PASSWORD = "test"
+    POSTGRES_DEFAULT_ROOT_USERNAME = "root"
+    POSTGRES_DEFAULT_DATABASE = "test"
+
+    attr_reader :port, :username, :password, :database
+
+    # Initializes a new instance of PostgresContainer
+    #
+    # @param image [String] the image to use
+    # @param username [String] the username to use
+    # @param password [String] the password to use
+    # @param database [String] the database to use
+    # @param port [String] the port to use
+    # @param kwargs [Hash] the options to pass to the container. See {DockerContainer#initialize}
+    # @return [PostgresContainer] a new instance of PostgresContainer
+    def initialize(image = POSTGRES_DEFAULT_IMAGE, username: nil, password: nil, database: nil, port: nil, **kwargs)
+      super(image, **kwargs)
+      @port = port || ENV.fetch("POSTGRES_PORT", POSTGRES_DEFAULT_PORT)
+      @username = username || ENV.fetch("POSTGRES_USER", POSTGRES_DEFAULT_USERNAME)
+      @password = password || ENV.fetch("POSTGRES_PASSWORD", POSTGRES_DEFAULT_PASSWORD)
+      @database = database || ENV.fetch("POSTGRES_DATABASE", POSTGRES_DEFAULT_DATABASE)
+    end
+
+    # Starts the container
+    #
+    # @return [PostgresContainer] self
+    def start
+      with_exposed_ports(@port)
+      _configure
+      super
+    end
+
+    # Returns the host used to connect to the container
+    # If the host is "localhost", it is replaced by "127.0.0.1" since Postgres fallbacks
+    # to a socket connection with "localhost"
+    #
+    # @return [String] the host used to connect to the container
+    def host
+      host = super
+      (host == "localhost") ? "127.0.0.1" : host
+    end
+
```

Comment:
```suggestion

```

This isn't needed for the Postgres client, when "localhost" is used it connects to the localhost instead of trying to connect to a socket